agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -494,34 +494,20 @@ func (state *DockerTaskEngineState) storeV3EndpointIDToDockerIDUnsafe(v3Endpoint
 	state.v3EndpointIDToDockerID[v3EndpointID] = dockerID
 }
 
-// ContainerByV3EndpointID returns an apicontainer.DockerContainer for a given v3 endpoint ID
-func (state *DockerTaskEngineState) ContainerByV3EndpointID(v3EndpointID string) (*apicontainer.DockerContainer, bool) {
+// DockerIDByV3EndpointID returns a docker ID for a given v3 endpoint ID
+func (state *DockerTaskEngineState) DockerIDByV3EndpointID(v3EndpointID string) (string, bool) {
 	state.lock.RLock()
 	defer state.lock.RUnlock()
 
 	dockerID, ok := state.v3EndpointIDToDockerID[v3EndpointID]
-	if !ok {
-		return nil, false
-	}
-
-	container, ok := state.idToContainer[dockerID]
-	if !ok {
-		return nil, false
-	}
-
-	return container, ok
+	return dockerID, ok
```

Comment:
this can just be `return  state.v3EndpointIDToDockerID[v3EndpointID]`

Reply:
When I do this, my IDE tells me `not enough arguments to return`, I think this is because 
you can use map in two ways: 
1. `dockerID := state.v3EndpointIDToDockerID[v3EndpointID]`
2. `dockerID, ok := state.v3EndpointIDToDockerID[v3EndpointID]`

It's unclear to the IDE (and programmers) when you just do `return state.v3EndpointIDToDockerID[v3EndpointID]`, so I prefer to keep it.

Same below.